tests/unit_tests/utils/test_model_data_writer.py

```diff
@@ -12,27 +13,29 @@
 logger.setLevel(logging.DEBUG)
 
 
-def test_write_metadata(args_dict_site, io_handler):
+def test_write_metadata(args_dict_site, io_handler, tmp_test_directory):
 
-    # test writer with no workflow configuration
-    w_1 = writer.ModelDataWriter()
-    with pytest.raises(
-        AttributeError, match=r"\'NoneType\' object has no attribute \'product_data_file_name\'"
-    ):
-        w_1.write_metadata()
+    # test writer of metadata
+    w_1 = writer.ModelDataWriter(args_dict=args_dict_site)
+    yml_file = w_1.write_metadata()
+    assert Path(yml_file).exists()
 
+    yml_file = w_1.write_metadata(ymlfile=str(tmp_test_directory) + "/test_file.yml")
```

Comment:
Should use `Path.join` or  the `\` notation here instead of adding strings

Reply:
OK!